Sprintfinity3902/Dungeon/Dungeon.cs

```diff
@@ -94,36 +118,14 @@ public IRoom GetById(int id)
 
         public void NextRoom()
         {
-            currentId = CurrentRoom.Id;
-            if(CurrentRoom.Id == 18)
-            {
-                CurrentRoom = GetById(1);
-            }
-            else
-            {
-                currentId++;
-                CurrentRoom = GetById(currentId);
-            }
-            NextId = CurrentRoom.Id;
-            CurrentRoom.garbage.Clear();
-            SetLinkPosition();
+            int currentId = (CurrentRoom.Id + 1) % 19 == 0 ? 1 : CurrentRoom.Id + 1;
+            SetCurrentRoom(currentId);
         }
 
         public void PreviousRoom()
         {
-            currentId = CurrentRoom.Id;
-            if (CurrentRoom.Id == 1)
-            {
-                CurrentRoom = GetById(18);
-            }
-            else
-            {
-                currentId--;
-                CurrentRoom = GetById(currentId);
-            }
-            NextId = CurrentRoom.Id;
-            CurrentRoom.garbage.Clear();
-            SetLinkPosition();
```

Comment:
This is the other half of the fix for the bug you have with going through doors right now. If you remove SetLinkPosition(); from SetCurrentRoom It needs to be called in NextRoom and PrevRoom

Reply:
Thanks, fixed this.